core/types/block_ext_test.go

```diff
@@ -106,3 +111,410 @@ func assertDifferentPointers[T any](t *testing.T, a *T, b any) {
+
+	withdrawals := []*ethtypes.Withdrawal{{Index: 9}} // ignored
+	block = block.WithWithdrawals(withdrawals)
+
+	extra := &BlockBodyExtra{
+		Version: 10,
+		ExtData: ptrTo([]byte{11}),
+	}
+	extras.Block.Set(block, extra)
+
+	b, err := rlp.EncodeToBytes(block)
+	require.NoError(t, err)
+
+	gotBlock := new(Block)
+	err = rlp.DecodeBytes(b, gotBlock)
+	require.NoError(t, err)
+
+	// Check transactions in the following because unexported fields
+	// size and time are set by the RLP decoder.
+	gotTxs := gotBlock.Transactions()
+	require.Len(t, gotTxs, 1)
+	gotTx := gotTxs[0]
+	gotTxBin, err := gotTx.MarshalBinary()
+	require.NoError(t, err)
+	wantTxBin, err := testTx.MarshalBinary()
+	require.NoError(t, err)
+	assert.Equal(t, wantTxBin, gotTxBin)
+
+	wantHeader := &Header{
+		ParentHash: common.Hash{1},
+		Extra:      []byte{}, // field set by RLP decoder
+	}
+	wantHeaderExtras := &HeaderExtra{ExtDataHash: common.Hash{2}}
+	SetHeaderExtra(wantHeader, wantHeaderExtras)
+
+	wantUncle := &Header{
+		ParentHash: common.Hash{6},
+		Difficulty: new(big.Int), // field set by RLP decoder
+		Number:     new(big.Int), // field set by RLP decoder
+		Extra:      []byte{},     // field set by RLP decoder
+	}
+	wantUncleExtra := &HeaderExtra{ExtDataHash: common.Hash{7}}
+	SetHeaderExtra(wantUncle, wantUncleExtra)
+	wantUncles := []*Header{wantUncle}
+
+	wantBlock := NewBlock(wantHeader, gotTxs /* checked above */, wantUncles, receipts, stubHasher{})
+	wantExtra := &BlockBodyExtra{
+		Version: 10,
+		ExtData: ptrTo([]byte{11}),
+	}
+	extras.Block.Set(wantBlock, wantExtra)
+	_ = wantBlock.Size() // set block cached unexported size field
+
+	assert.Equal(t, wantBlock, gotBlock)
+	gotExtra := extras.Block.Get(gotBlock)
+	assert.Equal(t, wantExtra, gotExtra)
+}
+
+// TestBlockBody tests the [BlockBodyExtra.Copy] method is implemented correctly.
+func TestBlockBody(t *testing.T) {
+	t.Parallel()
+
+	header := &Header{ParentHash: common.Hash{1}}
+	tx := NewTransaction(0, common.Address{1}, big.NewInt(2), 3, big.NewInt(4), []byte{5})
+	txs := []*Transaction{tx}
+	uncles := []*Header{{ParentHash: common.Hash{6}}}
+	receipts := []*Receipt{{PostState: []byte{7}}}
+	block := NewBlock(header, txs, uncles, receipts, stubHasher{})
+	blockExtras := &BlockBodyExtra{
+		Version: 8,
+		ExtData: ptrTo([]byte{9}),
+	}
+	allExportedFieldsSet(t, blockExtras) // make sure each field is checked
+	extras.Block.Set(block, blockExtras)
+
+	wantUncle := &Header{
+		ParentHash: common.Hash{6},
+		Difficulty: new(big.Int),
+		Number:     new(big.Int),
+	}
+	// uncle returned from Body() has its extra set, even if the block uncle did not have it set
+	SetHeaderExtra(wantUncle, &HeaderExtra{})
+	wantBody := &Body{
+		Transactions: []*Transaction{tx},
+		Uncles:       []*Header{wantUncle},
+	}
+	wantBodyExtra := &BlockBodyExtra{
+		Version: 8,
+		ExtData: ptrTo([]byte{9}),
+	}
+	extras.Body.Set(wantBody, wantBodyExtra)
+
+	body := block.Body()
```

Comment:
```suggestion
	body := block.Body() // this is expected to perform a deep copy
```
or something like that

Reply:
tests refactored, no need for this anymore.